src/instruction_generator/calls.rs

```diff
@@ -41,17 +41,17 @@ impl InstructionGenerator {
     ) {
         let Locatable { element: name, pos } = function_name;
         let qualified_name = name.demand_qualified();
-        let bare_name: &BareName = qualified_name.as_ref();
         // cloning to fight the borrow checker
         let function_parameters = self
-            .sub_program_parameters
-            .get_function_parameters(bare_name)
+            .subprogram_parameters
+            .get_function_parameters(&qualified_name)
             .clone();
         self.generate_push_named_args_instructions(&function_parameters, &args, pos);
-        self.push(Instruction::PushStack, pos);
+        let subprogram_name = SubprogramName::Function(qualified_name.clone());
```

Comment:
to avoid the cloning, if it's only for formatting, introduce structs like NameRef here